async.zsh

```diff
@@ -306,6 +306,22 @@ _async_zle_watcher() {
 	local worker=$ASYNC_PTYS[$1]
 	local callback=$ASYNC_CALLBACKS[$worker]
 
+	if [[ -n $2 ]]; then
+		# from man zshzle(1):
+		# `hup' for a disconnect, `nval' for a closed or otherwise
+		# invalid descriptor, or `err' for any other condition.
+		# Systems that support only the `select' system call always use
+		# `err'.
+
+		# this has the side effect to unregister the broken file descriptor
+		async_stop_worker $worker
+
+		if [[ -n $callback ]]; then
+			$callback '[async]' 2 "" 0 "$callback:zle -F $1 returned error $2" 0
```

Comment:
I think mentioning the `$worker` in the message would be good.

Reply:
you are right. Mentioning the worker makes more sense than the callback since it is the callback which receives the error message. Fixed.